plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/ConnectionHelper.java

```diff
@@ -79,21 +111,67 @@ public void validate() {
 				connectionState.remove(ConnectionState.VALID);
 			}
 
-			if (destination != null && destInterfaceList != null) {
+			if (destinationEndpoint != null && destInterfaceList != null) {
```

Comment:
Should this not rather just be:
```suggestion
			if (destInterfaceList != null) {
```
as above: https://github.com/eclipse-4diac/4diac-ide/blob/cd46a5449024fe2847367cd23856d8ba7f783ee7/plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/ConnectionHelper.java#L100-L101
or am I missing something?

Reply:
I think martin is right